main.go

```diff
@@ -29,12 +29,17 @@ import (
 
 	clientset "github.com/xmudrii/etcdproxy-controller/pkg/client/clientset/versioned"
 	informers "github.com/xmudrii/etcdproxy-controller/pkg/client/informers/externalversions"
+	"github.com/xmudrii/etcdproxy-controller/pkg/controller/etcdproxy"
 	"github.com/xmudrii/etcdproxy-controller/pkg/signals"
+	"github.com/xmudrii/etcdproxy-controller/pkg/utility"
 )
 
 var (
-	masterURL  string
 	kubeconfig string
+
+	etcdURL             string
+	etcdCAConfigMapName string
+	etcdCertSecretName  string
 )
 
 func main() {
```

Comment:
This main package has grown really large.  You should make a followup issue to refactor so the main package is small, gathers information from flags, then calls a different package to do the client building, informer starting, and wiring.

Reply:
Created issue #7 to track this.